packages/notifications/src/elements/toaster/ToastProvider.spec.tsx

```diff
@@ -34,33 +34,37 @@ const ToastExample: React.FC<IToastProviderProps> = props => {
 };
 
 describe('ToastProvider', () => {
-  it('renders toasts in their provided placement', () => {
+  const user = userEvent.setup();
+
+  it('renders toasts in their provided placement', async () => {
     const { getByRole, getByText } = render(<ToastExample />);
 
-    userEvent.click(getByRole('button'));
+    await user.click(getByRole('button'));
     const notificationElement = getByText('notification');
 
     expect(notificationElement).toBeInTheDocument();
     expect(notificationElement.parentElement?.parentElement).toHaveStyleRule('bottom', '64px');
     expect(notificationElement.parentElement?.parentElement).toHaveStyleRule('left', '12px');
   });
 
-  it('renders toasts in alternative placement when in RTL mode', () => {
+  it('renders toasts in alternative placement when in RTL mode', async () => {
     const { getByRole, getByText } = renderRtl(<ToastExample />);
 
-    userEvent.click(getByRole('button'));
+    await user.click(getByRole('button'));
     const notificationElement = getByText('notification');
 
     expect(notificationElement).toBeInTheDocument();
     expect(notificationElement.parentElement?.parentElement).toHaveStyleRule('bottom', '64px');
     expect(notificationElement.parentElement?.parentElement).toHaveStyleRule('right', '12px');
   });
 
-  it('renders visible toasts up to provided limit', () => {
+  it('renders visible toasts up to provided limit', async () => {
     const { getByRole, getAllByText } = render(<ToastExample />);
 
-    for (let x = 0; x < 10; x++) {
-      userEvent.click(getByRole('button'));
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    for (const _ of new Array(10).fill(null)) {
+      // eslint-disable-next-line no-await-in-loop
+      await user.click(getByRole('button'));
     }
```

Comment:
Should this be refactored as shown by https://eslint.org/docs/latest/rules/no-await-in-loop?